### PR TITLE
my dog ate my homework

### PR DESCRIPTION
**Pull request by F-Rivers**
# Submitting Week 4 Assessment

### ✅ Completed Files
Mark the following as confirmation that the files in this repository are complete.
- [ x] JavaScript Coding Challenges
- [ x] Ruby Coding Challenges
- [ x] Interview Questions

### 📝 Feedback
If you would like specific feedback on a particular problem please leave details here:

### CODE REVIEW

File: code-challenges.test.js [hunk 1]
```diff
@@ -15,35 +15,113 @@
...
+    const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre","aquamarine", "saffron"]
 // Expected output example (can be a different order): ["yellow", "blue", "pink", "green"]
-const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+        expect(shuffle(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+        expect(shuffle(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine","periwinkle", "indigo", "ochre"]))
+    })
+})
 
+// I got RED
```
Comment (SunkissedQueen) [inline code-challenges.test.js:32]: good failure

File: code-challenges.test.js [hunk 2]
```diff
@@ -15,35 +15,113 @@
...
 // b) Create the function that makes the test pass.
+// make a function called shuffle with 'array' as a parameter 
+// either .splice() or .slice() to remove a color??
+// use .sort() in conjunction with something to make it random 
+    // use math.random to randomize the array
+const shuffle = (array) => {
+    return array.slice(1).sort(() => Math.random() - 1)
+}
 
+// I got green
```
Comment (SunkissedQueen) [inline code-challenges.test.js:45]: green light!!!

File: code-challenges.test.js [hunk 3]
```diff
@@ -15,35 +15,113 @@
...
 
 // --------------------2) Create a function that takes an array of numbers and returns an array of the minimum and maximum numbers in that order.
 
 // a) Create a test with expect statements for each of the variables provided.
 
-const nums1 = [3, 56, 90, -8, 0, 23, 6]
+// the test should have the array of numbers in order from min to max 
+// expect() should contain the given expected output 
+
+describe("min_max", () => {
```
Comment (SunkissedQueen) [inline code-challenges.test.js:55]: good structure

File: code-challenges.test.js [hunk 4]
```diff
@@ -15,35 +15,113 @@
...
+
 // Expected output: [-8, 90]
-const nums2 = [109, 5, 9, 67, 8, 24]
+    expect(min_max(nums1)).toEqual([-8, 90])
 // Expected output: [5, 109]
+    expect(min_max(nums2)).toEqual([5, 109])
+         })
+})
 
+// I got RED
```
Comment (SunkissedQueen) [inline code-challenges.test.js:67]: thanks for putting test results

File: code-challenges.test.js [hunk 5]
```diff
@@ -15,35 +15,113 @@
...
+    let newArray = [Math.min(...array), Math.max(...array)]
+    return newArray.sort((a, b) => a - b)
+}
 
-// a) Create a test with an expect statement using the variables provided.
+// I got RED 
+//  ● min_max › takes an array of numbers, returns an array of the minimum and maximum numbers, in that order
+// ReferenceError: min_max is not defined
+
+// I got green but didn't change anything from when I got red??
```
Comment (SunkissedQueen) [inline code-challenges.test.js:87]: That's peculiar

File: code-challenges.test.js [hunk 6]
```diff
@@ -15,35 +15,113 @@
...
+    let combinedArray = [...array1, ...array2]
+    let noDoubles = combinedArray.filter((value, index) => index === combinedArray.indexOf(value))
+    return noDoubles
+}
+
+// I got red 
+//  ● doubles › takes in two arrays as arguments and returns one array with no duplicate values
+// ReferenceError: doubles is not defined
+
+// I got green but didn't change anything from when I got red??
```
Comment (SunkissedQueen) [inline code-challenges.test.js:127]: strange...everything passes on my side

File: interview-questions.md [hunk 7]
```diff
@@ -7,52 +7,64 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What are props in React?
 
-  Your answer:
+  Your answer: Props is how you pass information from the parent component to the child component. You make the information available by referencing the variable with props. 
 
-  Researched answer:
+  Researched answer: Props is a keyword in React, that stands for properties, and is being used for passing data from one component to another. The important part is that the data in props is being passed in a uni-directional flow (from parent to child). Props data is read-only, the data coming from the parent cannot be changed by the child component.
```
Comment (SunkissedQueen) [inline interview-questions.md]: Your opening looks very similar to  `“Props” is a special keyword in React, which stands for properties and is being used for passing data from one component to another. `

File: ruby-challenges.rb [hunk 8]
```diff
@@ -3,31 +3,96 @@
...
+        "#{number} is odd"
+    end 
+end 
+
+p even_or_odd(num1)
+p even_or_odd(num2)
+p even_or_odd(num3)
+p even_or_odd(3405)
+
+# output --->
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:30]: appreciate the output

File: ruby-challenges.rb [hunk 9]
```diff
@@ -3,31 +3,96 @@
...
 # Expected output: 'bby Rd'
 
+def no_vowels (words) 
+    "#{words}".delete "aeiouAEIOU"
+end
+
+p no_vowels(album1)
+p no_vowels(album2)
+p no_vowels(album3)
+p no_vowels("Frank The Tank")
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:56]: alright new test variables

File: ruby-challenges.rb [hunk 10]
```diff
@@ -3,31 +3,96 @@
...
+
 palindrome_tester1 = 'Racecar'
 # Expected output: 'Racecar is a palindrome'
 palindrome_tester2 = 'LEARN'
 # Expected output: 'LEARN is not a palindrome'
 palindrome_tester3 = 'Rotator'
 # Expected output: 'Rotator is a palindrome'
+
+
+def pali_word(word)
```
Comment (SunkissedQueen) [inline ruby-challenges.rb:81]: nice method

**Review by SunkissedQueen — DISMISSED** [review]
Frank, I will need you to go through your interview questions and ensure you have placed them in your own words. Since you will be sending changes, I recommend you include a definition, your use of the tool, and something interesting you learn about the tool. Let me know when you have completed this task so I can continue the review.

**Review by WalkerTexas-coder — CHANGES_REQUESTED** [review]
Frank I am putting in a request that you completely rewrite ALL of your answers and use absolutely zero language that comes from articles written by others. Copying and pasting your interview answers is not permitted even if we use proper citation. 

We are not here to test your ability to google but what you can explain in an interview. 

Your assessment will not be accepted until this is complete.

File: interview-questions.md [hunk 11]
```diff
@@ -7,53 +7,38 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  Your answer: A DOM event is when the user interacts with the web page. DOM stands for Document Object Model and we refer to it as just 'e'. 
 
-  source: https://www.w3schools.com/jsref/dom_obj_event.asp#:~:text=HTML%20DOM%20events%20allow%20JavaScript,a%20user%20clicks%20a%20button).
+  Researched answer: A DOM event is used in conjunction with functions that will not run until the DOM event is triggered. The events are written with the syntax of camelCase and written inside of curly brackets {}. One example of how we can trigger the DOM event is by clicking on an element in our page and this will generate an event such as, changing the info on the page or directing you elsewhere.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:20]: Better! I appreciate it a lot more being in your own words.

File: interview-questions.md [hunk 12]
```diff
@@ -7,53 +7,38 @@ Then, research the question to expand on your answer. Even if you feel you have
...
-  OOP - The code is centralized around instances, a person, a database connection, or an input field. All instances have a blueprint or 'classes'. While researching this, I found  interesting that the OOP approach is great for large, complex programs. 
-
-  FP - Unlike OOP, in functional programming we don’t have instances that hold their own state, it is actually what to avoid. There are no blueprints for instances. We factor down the task we want to implement into functions as much as possible. While researching this I found interesting that FP approach is great for in demand tasks such as data analysis. 
-
-  source: https://javascript.plainenglish.io/javascript-functional-vs-oop-fb5fbf15a35d
+  Your answer: OOP is a manner of of programming that focuses on objects, classes and, how to alter them to best fit our needs. Meanwhile, FP is focused on functions, logic and, are unable to alter them.   
 
+  Researched answer: Having a OOP style of programming means you're writing code in a manner that is focused on objects, hence the data being stored in objects. All focus points are a sort of blueprint, also known as a class. We use OOP in our class to make our code more dynamic.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:27]: YES!

File: interview-questions.md [hunk 13]
```diff
@@ -7,53 +7,38 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
 5. Ruby has an implicit return. What does that mean?
 
-  Your answer: Implicit return is when Ruby automatically knows to run the last code without being told to do so. In class we used the getter method to return the value of an attribute. We also used the setter method, which takes a parameter and assigns it to the attribute.
-
-  Researched answer: Any statement in Ruby returns the value of the last evaluated expression, we call a method and expect some value in return. Arrow functions ( => ) implicitly return values by not needing the curly braces {} that are supposed to wrap a function's body as long as their body only contains one expression.
+  Your answer: Implicit return is when Ruby automatically knows to run the last code of a method without being told to do so.
 
-  source: https://www.freecodecamp.org/news/idiomatic-ruby-writing-beautiful-code-6845c830c664/
+  Researched answer: Implicit return in Ruby is when the method being returned can run without being called upon by the "return" syntax.
```
Comment (WalkerTexas-coder) [inline interview-questions.md:41]: Variable not method but yes!

**Review by WalkerTexas-coder — APPROVED** [review]
Alright, Let's go!

**Pull request merged**